src/rest_utils.cpp

```diff
@@ -308,140 +317,56 @@ static Status parseOutputs(const ::KFSResponse& response_proto, rapidjson::Prett
         writer.EndArray();
         writer.Key("datatype");
         writer.String(tensor.datatype().c_str());
-        if (tensor.datatype() != "BYTES") {
+        bool binaryOutput = (binaryOutputs.count(tensor.name().c_str()) > 0) ? binaryOutputs[tensor.name().c_str()] : (tensor.datatype() == "BYTES");
```

Comment:
Don't use count when you only need to check for existence of such key, especially on critical path for rest handling. First improvement would be to use find() & comapare against the end of container, but event better would be using std::set instead of map -> it would allow for such error to occur.

Reply:
Also pass this container as const ref